File: 1929.Concatenation_of_Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */

// Solution: #1

var getConcatenation = function(nums) {
    return nums.concat(nums);
};


// Solution: #2
var getConcatenation = function(nums) {
    
    let newArray = new Array(nums.length * 2);

    for (let i = 0; i < nums.length; i++) {
        newArray[i] = newArray[i + nums.length] = nums[i];
    }

    return newArray;
};

File: Basics/1929.Concatenation_of_Array.cpp
// Solution: #1
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int inedex;
        vector<int> ans(nums.size() * 2);

        inedex = 0;
        for(auto i : nums) {
            ans[inedex] = i;
            ans[inedex + nums.size()] = i;
            inedex++;
        }

        return (ans);
    }
};

// Solution: #2
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans(nums.size() * 2);

        copy(nums.begin(), nums.end(), ans.begin());
        copy(nums.begin(), nums.end(), ans.begin() + nums.size());

        return (ans);
    }
};

// Solution: #2
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans(nums);

        for (auto i : nums) {
            ans.push_back(i);
        }

        return (ans);
    }
};